src/utils/functions.ts

```diff
@@ -0,0 +1,19 @@
+import { Status } from '../types/Status';
+import { Todo } from '../types/Todo';
+
+export function getFilteredTodos(todosList: Todo[], filterBy: Status) {
+  const filteredTodos = todosList.filter((todo: Todo) => {
+    switch (filterBy) {
+      case Status.Active:
+        return todo.completed === false;
+
+      case Status.Completed:
+        return todo.completed === true;
+
+      default:
+        return todo;
+    }
+  });
+
+  return filteredTodos;
+}
```

Comment:
We need to make a `filter` only if `filterBy` is not an empty string, so it is worth checking this condition first and only then make a `filter`, so as not to make unnecessary traversal of the array.
```suggestion
export function getFilteredTodos(todosList: Todo[], filterBy: Status) {
  if (filterBy === Status.Active) {
    todosList = todosList.filter((todo: Todo) => !todo.completed).
  }
  if (filterBy === Status.Completed) {
    todosList = todosList.filter((todo: Todo) => todo.completed).
  }

  return todosList;
}
```